File: Day15/com/test/score.py
class Score:
    def __init__(self, no, name, kor, eng, mat):
        self.__no = no
        self.__name = name
        self.__kor = kor
        self.__eng = eng
        self.__mat = mat


    def setNo(self, no):
        self.__no = no
    def setName(self, name):
        self.__name = name
    def setK(self, kor):
        self.__kor = kor
    def setE(self, eng):
        self.__eng = eng
    def setM(self, mat):
        self.__mat = mat


    def getNo(self):
        return self.__no
    def getName(self):
        return self.__name
    def getK(self):
        return self.__kor
    def getE(self):
        return self.__eng
    def getM(self):
        return self.__mat
    def tot(self):
        return self.getK() + self.getE() + self.getM()
    def avg(self):
        return self.tot() // 3
    def grade(self):
        if self.avg() >= 90:
            return "A"
        elif 80 <= self.avg() < 90:
            return "B"
        elif 70 <= self.avg() < 80:
            return "C"
        else:
            return "F"
    def __str__(self):
        return "No: %d, Name: %s, Kor: %d, Eng: %d, Mat: %d, Tot: %d, Avg: %d, Grade: %s"\
              %(self.getNo(), self.getName(), self.getK(), self.getE(), self.getM(), self.tot(), self.avg(), self.grade())

if __name__ == '__main__':
    s1 = Score(1."홍길동",90,100,80)
    print(s1)

File: Day54_java08/src/com/exam/Score.java
package com.exam;

// 3과목 총점 평균 학점
public class Score {

		private String name;
		private int kor; 
		private int eng;
		private int mat;
				
		public Score() {
			super();			
		}
		
		public Score(String name, int kor, int eng, int mat) {
			super();
			this.name = name;
			this.kor = kor;
			this.eng = eng;
			this.mat = mat;
		}
		
		
		public String getName() {
			return name;
		}

		public void setName(String name) {
			this.name = name;
		}

		public int getKor() {
			return kor;
		}
		public void setKor(int kor) {
			this.kor = kor;
		}
		public int getEng() {
			return eng;
		}
		public void setEng(int eng) {
			this.eng = eng;
		}
		public int getMat() {
			return mat;
		}
		public void setMat(int mat) {
			this.mat = mat;
		}
		
		public int getTot() {
			int tot = 0;
			tot = this.getKor() + this.getEng() + this.getMat();
			return tot;
		}
		
		
		public double getAvg() {
			double avg =0;
			avg = this.getTot()/3.0;
			return avg;
			
		}

		public String Getgrade() {
			String grade = null; //메서드안에 초기변수를 주면 지역변수?
			switch((int)getAvg()/10) { //98.0 --> 9.8 -> 9 / 100.0 -> 10.0 -> 10
			case 10:
			case 9 : grade = "A";
					 break;
			case 8 : grade = "B";
					 break;
			case 7 : grade = "C";
					 break;
			case 6 : grade = "D";
					 break;
		    default : 
		    		grade = "F";
			}
			return grade;
			}
		
		@Override
		public String toString() {
			String str = String.format("%10s %5d %5d %5d %5d %3.1f %3s\n", 
					this.getName(), this.getKor(), this.getEng(), this.getMat(), this.getTot(),this.getAvg(),this.Getgrade());
			return str;
		}

	}


	
